tests/basic-test.sh

```diff
@@ -41,6 +41,7 @@ fi
 validate_checkout_basic() {
     (cd $1;
      assert_has_file firstfile
+     assert_not_streq $(stat -c '%h' firstfile) 1
```

Comment:
Wait, I read this backwards. The fact that it's failing means that it *is* copying rather than hardlinking, which is what we want in user-mode without `-U`. So I think this just needs some conditionals around it.